File: src/main/java/io/github/saneea/dvh/feature/print/PrintArgsFeature.java
package io.github.saneea.dvh.feature.print;

import java.io.PrintStream;

import io.github.saneea.dvh.Feature;
import io.github.saneea.dvh.FeatureContext;

public class PrintArgsFeature implements Feature, Feature.Out.Text.PrintStream {

	private PrintStream out;

	@Override
	public Meta meta(FeatureContext context) {
		return Meta.from("print CLI args");
	}

	@Override
	public void run(FeatureContext context) throws Exception {
		for (String arg : context.getArgs()) {
			out.println(arg);
		}
	}

	@Override
	public void setOut(PrintStream out) {
		this.out = out;
	}

}

File: src/main/java/io/github/saneea/dvh/feature/print/PrintArgsFeature.kt
package io.github.saneea.dvh.feature.print

import io.github.saneea.dvh.Feature
import io.github.saneea.dvh.Feature.Meta
import io.github.saneea.dvh.FeatureContext
import java.io.PrintStream

class PrintArgsFeature :
    Feature,
    Feature.Out.Text.PrintStream {

    override lateinit var context: FeatureContext
    override lateinit var outTextPrintStream: PrintStream

    override val meta get() = Meta("print CLI args")

    override fun run() =
        context.args.forEach(outTextPrintStream::println)
}